src/org/mozartoz/truffle/nodes/control/AndThenNode.java

```diff
@@ -0,0 +1,23 @@
+package org.mozartoz.truffle.nodes.control;
+
+import org.mozartoz.truffle.nodes.OzNode;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+
+public class AndThenNode extends OzNode {
+	private @Child OzNode left, right;
+
+	private final ConditionProfile conditionProfile = ConditionProfile.createCountingProfile();
+
+	public AndThenNode(OzNode left, OzNode right) {
+		this.left = left;
+		this.right = right;
```

Comment:
You need to wrap these with `DerefNode.create(node)` as you use the deref'd value below.